src/DMXSerial2.cpp

```diff
@@ -829,7 +851,95 @@ void DMXSerialClass2::_processRDMMessage(byte CmdClass, uint16_t Parameter, bool
+              _rdm.packet.Data[4] = E120_CC_GET;
+            else if (!_initData->parameters[parameterNr].getSupported && _initData->parameters[parameterNr].setSupported)
+              _rdm.packet.Data[4] = E120_CC_SET;
+            else if (_initData->parameters[parameterNr].getSupported && _initData->parameters[parameterNr].setSupported)
+              _rdm.packet.Data[4] = E120_CC_GET_SET;
+            _rdm.packet.Data[5] = 0x00;
+            _rdm.packet.Data[6] = _initData->parameters[parameterNr].unit;
+            _rdm.packet.Data[7] = _initData->parameters[parameterNr].prefix;
+            WRITELONG(_rdm.packet.Data +  8, _initData->parameters[parameterNr].rangeMin);
+            WRITELONG(_rdm.packet.Data + 12, _initData->parameters[parameterNr].rangeMax);
+            WRITELONG(_rdm.packet.Data + 16, _initData->parameters[parameterNr].defaultValue);
+            memcpy(_rdm.packet.Data + 20, _initData->parameters[parameterNr].description, _rdm.packet.DataLength - 20);
+            handled = true;
+            ///////////////////////////////////////////
+          }
+        }
+      } else if (CmdClass == E120_SET_COMMAND) {
+        // Unexpected set
+        nackReason = E120_NR_UNSUPPORTED_COMMAND_CLASS;
+      }
+
+    } else if (Parameter == SWAPINT(E120_SENSOR_VALUE) && _initData->sensorsLength > 0) { // 0x0201
+      if (CmdClass == E120_GET_COMMAND) {
+        if (_rdm.packet.DataLength != 1) {
+          // Unexpected data
+          nackReason = E120_NR_FORMAT_ERROR;
+        } else if (_rdm.packet.SubDev != 0) {
+          // No sub-devices supported
+          nackReason = E120_NR_SUB_DEVICE_OUT_OF_RANGE;
+        } else {
+          uint8_t sensorNr = _rdm.packet.Data[0];
+          if (sensorNr >= _initData->sensorsLength) {
+            // Out of range sensor
+            nackReason = E120_NR_DATA_OUT_OF_RANGE;
+          } else {
+            int16_t sensorValue = 0;
+            int16_t lowestValue = 0;
+            int16_t highestValue = 0;
+            int16_t recordedValue = 0;
+            bool8 res = false;
+            if (_sensorFunc) {
+              res = _sensorFunc(sensorNr, &sensorValue, &lowestValue, &highestValue, &recordedValue);
+            }
+            if (res) {
+              _rdm.packet.DataLength = 9;
+              _rdm.packet.Data[0] = sensorNr;
+              WRITEINT(_rdm.packet.Data +  1, sensorValue);
+              WRITEINT(_rdm.packet.Data +  3, lowestValue);
+              WRITEINT(_rdm.packet.Data +  5, highestValue);
+              WRITEINT(_rdm.packet.Data +  7, recordedValue);
+              handled = true;
+            } else {
+              nackReason = E120_NR_HARDWARE_FAULT;
+            }
+          }
+        }
+      } else if (CmdClass == E120_SET_COMMAND) {
+        // Unhandled set. Set on a sensor is used to reset stats.
+        // User should process it in own handler when sensor supports high/low or recorded value.
```

Comment:
Given low/high and recorded is passed into the library, doing it centrally would be nicer, along with RECORD_SENSORS PID.